pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test: TestLogic_advisory_lock failed

pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test.TestLogic_advisory_lock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21324643?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21324643?buildTab=artifacts#/) on master @ [d1453c68043c6565020358b48eccdca76b408bc2](https://github.com/cockroachdb/cockroach/commits/d1453c68043c6565020358b48eccdca76b408bc2):


```
=== RUN   TestLogic_advisory_lock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/8ee417166775d051de542acde5a083e2/logTestLogic_advisory_lock2078211271
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156124 (expected label: C-bug)
I260423 16:32:41.301576 1 (gostd) rand.go:288  [-] 1  random seed: -6343106583402836279
[16:32:47] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19065/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/advisory_lock with config fakedist-vec-off: 71 tests, 2 failures
[16:32:48] --- total progress: 71 statements
--- total: 71 tests, 2 failures
    logic.go:4858: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8ee417166775d051de542acde5a083e2/logTestLogic_advisory_lock2078211271
--- FAIL: TestLogic_advisory_lock (7.26s)
=== RUN   TestLogic_advisory_lock/advisory_try_exclusive_vs_shared_holder
    logic.go:3264: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19065/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/advisory_lock:344: SELECT pg_try_advisory_xact_lock(702345678)
        expected:
            true
            
        but found (query options: "") :
            false
            
    logic.go:2420: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19065/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/advisory_lock:349: too many errors encountered, skipping the rest of the input
--- FAIL: TestLogic_advisory_lock/advisory_try_exclusive_vs_shared_holder (0.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_advisory_lock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63201

pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test.TestLogic_advisory_lock [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21325163?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21325163?buildTab=artifacts#/) on master @ [28a853eb4b32b35fea59230757a64bcc8ed1f40b](https://github.com/cockroachdb/cockroach/commits/28a853eb4b32b35fea59230757a64bcc8ed1f40b):


```
=== RUN   TestLogic_advisory_lock
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/8ee417166775d051de542acde5a083e2/logTestLogic_advisory_lock2705608561
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156124 (expected label: C-bug)
[21:57:27] setting distsql_workmem='28212B';
[21:57:27] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18960/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/advisory_lock with config fakedist-vec-off: 27 tests, 2 failures
[21:57:28] --- total progress: 27 statements
--- total: 27 tests, 2 failures
    logic.go:4858: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8ee417166775d051de542acde5a083e2/logTestLogic_advisory_lock2705608561
--- FAIL: TestLogic_advisory_lock (3.40s)
=== RUN   TestLogic_advisory_lock/pg_try_advisory_xact_lock_conflict
    logic.go:3264: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18960/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/advisory_lock:115: SELECT pg_try_advisory_xact_lock(4242424242)
        expected:
            true
            
        but found (query options: "") :
            false
            
    logic.go:2420: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18960/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test_/fakedist-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/advisory_lock:120: too many errors encountered, skipping the rest of the input
--- FAIL: TestLogic_advisory_lock/pg_try_advisory_xact_lock_conflict (0.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_advisory_lock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/logictest/testdata/logic_test/advisory_lock
+++ b/pkg/sql/logictest/testdata/logic_test/advisory_lock
@@ -111,6 +111,7 @@ COMMIT
 
 user testuser
 
+retry
 query B
 SELECT pg_try_advisory_xact_lock(4242424242)
 ----
@@ -340,6 +341,9 @@ COMMIT
 
 user testuser
 
+# Intent resolution is asynchronous, so after the commit it may be possible
+# that the lock is still held by the other session.
+retry
 query B
 SELECT pg_try_advisory_xact_lock(702345678)
 ----
@@ -377,6 +381,9 @@ COMMIT
 
 user testuser
 
+# Intent resolution is asynchronous, so after the commit it may be possible
+# that the lock is still held by the other session.
+retry
 query B
 SELECT pg_try_advisory_xact_lock_shared(703456789)
 ----
@@ -405,6 +412,7 @@ user testuser
 statement ok
 BEGIN
 
+retry
 query B
 SELECT pg_try_advisory_xact_lock(705678901)
 ----

